### ISSUE TITLE
Update service base image to 5.3

### ISSUE THREAD

**Issue by finestructure** [SwiftPackageIndex/SwiftPackageIndex-Server#731]
We're still running swift 5.2.4 and embedding an additional 5.3 swift toolchain to be able to dump tools version 5.3 packages. Since 5.3 is officially released, we could move the whole service to 5.3.

However, the service doesn't compile on 5.3. The issue is Ink doesn't seem to be compatible with 5.3 on Linux, which is also showing in our build matrix: https://swiftpackageindex.com/builds/3D9E38AF-2F20-4C81-AE8C-AAB59CFA03B4

We're getting the same error when compiling the server:

```
swift: /home/buildnode/jenkins/workspace/oss-swift-5.3-package-linux-ubuntu-18_04/swift/lib/AST/Expr.cpp:1980: swift::Expr *swift::AutoClosureExpr::getUnwrappedCurryThunkExpr() const: Assertion `false && "Malformed curry thunk?"' failed.
Stack dump:
0.      Program arguments: /usr/bin/swift -frontend -c -primary-file /host/.build/checkouts/Ink/Sources/Ink/API/Markdown.swift -primary-file /host/.build/checkouts/Ink/Sources/Ink/API/MarkdownParser.swift -primary-file /host/.build/checkouts/Ink/Sources/Ink/API/Modifier.swift -primary-file /host/.build/checkouts/Ink/Sources/Ink/Internal/Blockquote.swift -primary-file /host/.build/checkouts/Ink/Sources/Ink/Internal/Character+Classification.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/Character+Escaping.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/CodeBlock.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/FormattedText.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/Fragment.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/HTML.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/HTMLConvertible.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/Hashable+AnyOf.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/Heading.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/HorizontalLine.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/Image.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/InlineCode.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/KeyPathPatterns.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/Link.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/List.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/Metadata.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/Modifiable.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/ModifierCollection.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/NamedURLCollection.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/Paragraph.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/PlainTextConvertible.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/Readable.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/Reader.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/Require.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/Substring+Trimming.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/Table.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/TextStyle.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/TextStyleMarker.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/URL.swift /host/.build/checkouts/Ink/Sources/Ink/Internal/URLDeclaration.swift -supplementary-output-file-map /tmp/supplementaryOutputs-a57391 -target x86_64-unknown-linux-gnu -disable-objc-interop -I /host/.build/x86_64-unknown-linux-gnu/debug -enable-testing -g -module-cache-path /host/.build/x86_64-unknown-linux-gnu/debug/ModuleCache -profile-generate -profile-coverage-mapping -swift-version 5 -Onone -D SWIFT_PACKAGE -D DEBUG -enable-anonymous-context-mangled-names -parse-as-library -module-name Ink -o /host/.build/x86_64-unknown-linux-gnu/debug/Ink.build/API/Markdown.swift.o -o /host/.build/x86_64-unknown-linux-gnu/debug/Ink.build/API/MarkdownParser.swift.o -o /host/.build/x86_64-unknown-linux-gnu/debug/Ink.build/API/Modifier.swift.o -o /host/.build/x86_64-unknown-linux-gnu/debug/Ink.build/Internal/Blockquote.swift.o -o /host/.build/x86_64-unknown-linux-gnu/debug/Ink.build/Internal/Character+Classification.swift.o -index-store-path /host/.build/x86_64-unknown-linux-gnu/debug/index/store -index-system-modules 
1.      Swift version 5.3 (swift-5.3-RELEASE)
2.      While walking into decl 'MarkdownParser' (at /host/.build/checkouts/Ink/Sources/Ink/API/MarkdownParser.swift:17:8)
3.      While walking into body of 'parse(_:)' (at /host/.build/checkouts/Ink/Sources/Ink/API/MarkdownParser.swift:42:12)
4.      While walking into initializer for declaration 0x9545490 (at /host/.build/checkouts/Ink/Sources/Ink/API/MarkdownParser.swift:70:17)
```

**Comment by finestructure** [comment]
I'll see if I can fix and upstream this.

**Comment by finestructure** [comment]
Actually, that's a compiler crash, not a build error. I'll raise an issue.

**Comment by finestructure** [comment]
https://bugs.swift.org/browse/SR-13645

**Comment by finestructure** [comment]
FYI @johnsundell

**Comment by JohnSundell** [comment]
Would love to merge a fix/workaround for this if you happen to find one. I'm not using Ink on Linux myself, so I'm unlikely to be able to find the time to work on that.

**Comment by finestructure** [comment]
Huh, I actually thought there isn't much I could do here but it turns out I may have found a simple workaround 🙂

https://github.com/JohnSundell/Ink/pull/59

**Comment by finestructure** [comment]
The crash has been marked as resolved and available in 5.3 snapshots 🎉 I guess it'll be a while until that shows up in any releases, though.

@JohnSundell , were you planning to push an Ink 0.5.1 release? Anything I can do to help? Would be great if I could avoid switching to a fork :)

**Comment by finestructure** [comment]
This has been fixed by pinning to an explicit Ink version.
